contents/tutorials/vue-ab-tests.md

```diff
@@ -0,0 +1,180 @@
+
+Finally, activate your plugin in `main.js`:
+
+```js file=main.js
+import { createApp } from 'vue'
+import App from './App.vue'
+import posthogPlugin from '../plugins/posthog';
+
+const app = createApp(App);
+app.use(posthogPlugin);
+app.mount('#app')
+```
+
+Once you’ve done this, reload your app and click the button a few times. You should see events appearing in the [PostHog events explorer](https://app.posthog.com/events).
+
+![Events](../images/tutorials/vue-ab-tests/events.png)
+
+## Creating a custom event for our A/B test goal
+
+The first part of setting up our A/B test in PostHog is setting up the goal. We'll use the number of clicks on the button as our goal.
+
+To measure this, we [capture a custom event](/docs/product-analytics/capture-events) `home_button_clicked` when the button is clicked. To do this, update `handleClick` function in `App.vue` with the following:
+
+```vue file=App.vue
+<!-- rest of your code -->
+  methods: {
+    handleClick() {
+      this.$posthog.capture('home_button_click');
+    }
+  }
+<!-- rest of your code -->
+```
+
+## Creating an A/B test in PostHog
+
+The next step is to set up an A/B test (we call them experiments in PostHog).
+
+Go to the [A/B testing tab](https://app.posthog.com/experiments) in PostHog and click "New experiment". Add the following details to your experiment:
+
+1. Name it "Vue button text experiment".
+2. Set "Feature flag key" to `vue-button-text-experiment`.
+3. Under the experiment goal, select the `home_button_clicked` event we created in the previous step.
+4. Use the default values for all other fields.
+
+Click "Save as draft" and then click "Launch".
+
+![Vue.js experiment setup in PostHog](../images/tutorials/vue-ab-tests/new-experiment.png)
+
+## Implementing the A/B test code
+
+To implement the A/B test, we fetch the `vue-button-text-experiment` feature flag and update the button text based on whether the user is in the `control` or `test` variant of the experiment.
+
+Update your code in `App.vue` to the following:
+
+```vue file=App.vue
+<template>
+  <div id="app">
+    <h1>Vue.js A/B test</h1>
+    <button @click="handleClick">{{ buttonText }}</button>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'App',
+  data() {
+    return {
+      buttonText: 'No Variant'
+    };
+  },
+  mounted() {
+    const posthog = this.$posthog;
+    if (posthog.getFeatureFlag('vue-button-text-experiment') === 'control') {
+      this.buttonText = 'Control variant';
+    } else if (posthog.getFeatureFlag('vue-button-text-experiment') === 'test') {
+      this.buttonText = 'Test variant';
+    }
+  },
+  methods: {
+    handleClick() {
+      this.$posthog.capture('home_button_click');
+    }
+  }
+}
+</script>
+```
+
+Now if you refresh your app, you should see the button text updated to either `Control variant` or `Test variant`. 
```

Comment:
This seems to load "no variant" on first load for some reason, maybe isn't checking PostHog on mount? Works fine on subsequent loads.